Oleksandr week 1 react assignment

aleksandrvasilyev opened the pull request.

robvk reviewed (changes requested):
A good start! Some general comments:
- I like your component splitting, seems very logical!
- If React doesn't need to 'react' to something then it should not be in the state. Remember that every change in state causes a rerender so we want to minimize it if possible
- In the [example](https://hyf-react-w1-example.netlify.app/) I can `deselect` a category by clicking on it a second time to get the full list again, but I can't in yours. Can you look into this and see if you can add that feature?

Good luck!

aleksandrvasilyev commented:
Hello Rob, thank you for the feedback. I removed unnecessary states and added the `deselect` category feature. Should I split components further into `Product` and `Category`?

robvk commented:
Nice work! Good luck with week 2
